### PR TITLE
Major endpoint

### LINKED ISSUE #8: [P0] Add Major Model & Endpoints
Endpoints:
- Route `/majors`
   - GET - list of all majors
- Route `/majors/<id>`
   - GET - a major of the specified ID

Model:
```
name: string
courses: Course[]
desc?: string
college: string
```

### PR DESCRIPTION
**Pull request by 24mahik**
close #8

### CODE REVIEW

**Review by SirajChokshi — APPROVED** [review]
good work!

File: api/routes/reviews.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+});
+
+// Getting one review by id
+router.get('/:id', getReview, (req, res) => {
+  res.json(res.review);
+});
+
+// update review
+router.patch('/:id', getReview, async (req, res) => {
+  if (req.body.title != null) {
```
Comment (daviskeene) [inline api/routes/reviews.js:43]: I think this is fine, not sure if there's an easier way to update several fields in JavaScript but this gets the job done.

File: api/routes/reviews.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    res.status(400).json({ message: err.message });
+  }
+});
+
+// get review object by ID
+async function getReview(req, res, next) {
+  try {
+    review = await Review.findById(req.params.id);
+    if (review == null) {
+      return res.status(404).json({ message: 'Cant find review' });
```
Comment (daviskeene) [inline api/routes/reviews.js:74]: Might be helpful to include the failed review ID in this error message:

Comment (daviskeene) [reply api/routes/reviews.js:74]: `Can't find review with ID 12345`

**Review by daviskeene — APPROVED** [review]
LGTM! A few nits here and there, but these are merely suggestions.

**Pull request merged**